Library/OcAppleKernelLib/PrelinkedContext.c

```diff
@@ -1256,6 +1389,106 @@ PrelinkedInjectKext (
+  IN     CONST UINT8        *Executable OPTIONAL,
+  IN     UINT32             ExecutableSize OPTIONAL,
+  OUT    CHAR8              BundleVersion[MAX_INFO_BUNDLE_VERSION_KEY_SIZE] OPTIONAL
+  )
+{
+  EFI_STATUS           Status;
+  CHAR16               EFIVarName[64];
+  VOID                 *Buffer;
+  UINT32               EntryLength;
+  LILU_INJECTION_INFO  *InjectionInfo;
+  UINT32               InfoPlistOffset;
+  UINT32               ExecutableOffset;
+  UINTN                LiluInjectionInfoAddr;
+  UINT64               LiluInjectionInfoAddr64;
+  EFI_GUID             Guid = OC_READ_ONLY_VARIABLE_GUID;
+
+  EntryLength      = sizeof (LILU_INJECTION_INFO) + InfoPlistSize + ExecutableSize;
+  InfoPlistOffset  = sizeof (LILU_INJECTION_INFO);
+  ExecutableOffset = InfoPlistOffset + InfoPlistSize;
+
+  // Allocate buffer and setup header
+  LiluInjectionInfoAddr = (UINTN)AllocateRuntimePool (EntryLength);
+
+  Buffer = (VOID *)LiluInjectionInfoAddr;
+  if (!Buffer) {
+    return EFI_OUT_OF_RESOURCES;
+  }
+
+  InjectionInfo              = (LILU_INJECTION_INFO *)Buffer;
+  InjectionInfo->Version     = 0;
+  InjectionInfo->EntryLength = EntryLength;
+  if (AsciiStrCpyS (InjectionInfo->BundlePath, sizeof (InjectionInfo->BundlePath), BundlePath) != RETURN_SUCCESS) {
+    FreePool (Buffer);
+    return EFI_UNSUPPORTED;
+  }
+
+  InjectionInfo->KCKind          = KCKind;
+  InjectionInfo->InfoPlistOffset = InfoPlistOffset;
+  InjectionInfo->InfoPlistSize   = InfoPlistSize;
+
+  if (Executable != NULL) {
+    if (AsciiStrCpyS (InjectionInfo->ExecutablePath, sizeof (InjectionInfo->ExecutablePath), ExecutablePath) != RETURN_SUCCESS) {
+      FreePool (Buffer);
+      return EFI_UNSUPPORTED;
+    }
+
+    InjectionInfo->ExecutableOffset = ExecutableOffset;
+    InjectionInfo->ExecutableSize   = ExecutableSize;
+  } else {
+    InjectionInfo->ExecutableOffset = 0;
+    InjectionInfo->ExecutableSize   = 0;
+  }
+
+  CopyMem ((UINT8 *)Buffer + InfoPlistOffset, InfoPlist, InfoPlistSize);
+  if (Executable != NULL) {
+    CopyMem ((UINT8 *)Buffer + ExecutableOffset, Executable, ExecutableSize);
+  }
+
+  // Set EFIVarName
+  Status = OcUnicodeSafeSPrint (
+             EFIVarName,
+             sizeof (EFIVarName),
+             L"lilu-injection-info-addr-%d",
+             Context->LiluKextCount
+             );
+  if (EFI_ERROR (Status)) {
+    FreePool (Buffer);
+    return EFI_INVALID_PARAMETER;
+  }
+
+  LiluInjectionInfoAddr64 = (UINT64)LiluInjectionInfoAddr;
+  Status                  = OcSetSystemVariable (
+                              EFIVarName,
+                              OPEN_CORE_NVRAM_ATTR,
+                              8,
+                              &LiluInjectionInfoAddr64,
+                              &Guid
+                              );
+  if (EFI_ERROR (Status)) {
+    return Status;
+  }
+
+  Context->LiluKextCount++;
+
+  // To make caller of the function happy
+  if (BundleVersion != NULL) {
+    BundleVersion = "1.0.0";
```

Comment:
Overwriting a local variable will make neither the caller nor me happy. :)

Reply:
> Overwriting a local variable will make neither the caller nor me happy. :)

This is doing nothing atm right @mhaeuser? Should I omit it or overwrite the contents in some way?

Reply:
You can omit it, doesn't really matter: https://github.com/acidanthera/OpenCorePkg/blob/0097c13e686186d876b6208a1f2851454230764f/Library/OcMainLib/OpenCoreKernel.c#L589

Reply:
Done